### PR TITLE
DHFPROD-6247: Add sourceName to records during ingestion and make it facetable

### PR DESCRIPTION
**Pull request by rahulvudutala**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/test/java/com/marklogic/hub/step/impl/IngestWithSourceNameAndSourceTypeTest.java [hunk 1]
```diff
@@ -53,8 +53,8 @@ void runDefaultIngestionWithSourceNameAndSourceType(){
         assertEquals(JobStatus.FINISHED.toString(), response.getJobStatus());
         JsonNode sourceDoc = getStagingDoc("customer1.json");
         assertNotNull(sourceDoc);
-        JsonNode sourcesNode = sourceDoc.get("envelope").get("headers").get("sources").get(0);
-        assertEquals("sample-source", sourcesNode.get("name").asText());
+        JsonNode sourcesNode = sourceDoc.get("envelope").get("headers").get("sources").get(1);
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/java/com/marklogic/hub/step/impl/IngestWithSourceNameAndSourceTypeTest.java:56]: Can you add another assertion on "sources[0]", verifying that it's the step name?

**Review by rjrudin — CHANGES_REQUESTED** [review]
Looks good, just one tiny test addition - thanks for taking care of this!

**Review by rjrudin — DISMISSED** [review]

**Review by xnikhil08 — APPROVED** [review]

**Review by timur-isangulov — APPROVED** [review]

**Comment by rahulvudutala** [comment]
Made changes to hc-qa-project and fixed e2e tests

**Review by rjrudin — DISMISSED** [review]

**Review by xnikhil08 — DISMISSED** [review]

**Review by ryanjdew — DISMISSED** [review]

**Comment by rahulvudutala** [comment]
Changed test data for failing calculateMergingActivity.sjs test

**Review by ryanjdew — APPROVED** [review]

**Review by Sanjeevani19 — APPROVED** [review]

**Pull request merged**